### PR TITLE
test(pi-cli): lock in session create defaults contract (#70)

### LINKED ISSUE #70: bug: TestRunSessionCreateMissingRequiredFields fails on main
## Description

`TestRunSessionCreateMissingRequiredFields` in `skills/pi-rpc/scripts/cmd/pi-cli` fails on `main`. The test expects an error when `provider` or `model` is not set, but the validation is not implemented (or has regressed).

```
--- FAIL: TestRunSessionCreateMissingRequiredFields (0.20s)
    session_test.go:83: expected error when provider/model not set
FAIL    github.com/nq-rdl/agent-skills/skills/pi-rpc/scripts/cmd/pi-cli    0.234s
```

## Discovery

Surfaced when attempting to push a Makefile-only fix (PR #69). The pre-push hook runs `go test -race ./...`, which caught the failure. The fix in PR #69 was pushed with `--no-verify` because the test failure pre-dates those changes (verified by stashing and running the test against unmodified `main`).

## Impact

- The pre-push hook is silently bypassed by anyone who sees this failure and uses `--no-verify`
- Required-field validation for `pi-cli session create` may not be enforced at runtime

## Steps to reproduce

```bash
cd skills/pi-rpc/scripts
go test ./cmd/pi-cli/... -run TestRunSessionCreateMissingRequiredFields -v
```

## Expected

Test passes — `pi-cli session create` returns an error when `--provider` or `--model` is omitted.

## Suggested fix

Check `session_test.go:83` — either the validation was removed from the `create` command handler and needs to be restored, or the test expectation no longer matches the current CLI flag structure.

### PR DESCRIPTION
**Pull request by rudolfjs**
## Closes #70 — stacked on #69

This PR adds the positive test counterpart to `TestRunSessionCreateUnreachableServer` introduced in #69.

### What it does
- **Adds `TestRunSessionCreateDefaultsAccepted`** in `skills/pi-rpc/scripts/cmd/pi-cli/session_test.go`
- Spins up an `httptest` server that **decodes the Create request body** and asserts the CLI forwards empty `provider` / `model` **verbatim**, so the server can apply `PI_DEFAULT_PROVIDER` / `PI_DEFAULT_MODEL`.
- Adds a changie fragment.

### Why two tests?
After #69 the CLI's Create behavior is pinned from both sides:

| Scenario | Test | Outcome |
|---|---|---|
| Server unreachable, empty flags | `TestRunSessionCreateUnreachableServer` | RPC error surfaces to user |
| Server reachable, empty flags | `TestRunSessionCreateDefaultsAccepted` (new) | 200 OK + empty strings forwarded verbatim |

Together these prevent the #70 regression mode where a test passed *incidentally* (`err != nil` from a connection failure because no `pi-server` was running locally).

### Stacking notes
- Base: `fix/pi-rpc-cross-compile` (PR #69)
- Head: `bug/testrunmissing-pi-rpc`
- If #69 rebases, this branch will need a rebase too; merge-order is #69 → this PR.

### Verification
```
go test -race -count=1 ./skills/pi-rpc/scripts/cmd/pi-cli/...
ok  	github.com/nq-rdl/agent-skills/skills/pi-rpc/scripts/cmd/pi-cli	1.032s
```

### Hook bypass disclosure
Push used `--no-verify` (with user authorization) to route around a pre-existing flaky `go-test-jules` test (`TestOrchestratorWorkflowIntegration`: `fatal: a branch named 'main' already exists`). That flake is unrelated to this change and reproduces only under lefthook's parallel runner; solo runs are green.

### Review provenance
Plan was dual-reviewed (`gemini-3.1-pro-preview` + `pi --provider openai-codex --model gpt-5.4:xhigh`). Both converged on: keep #69's negative test, add the positive test with **request-body assertion** on empty provider/model forwarding (what this PR does).

### CODE REVIEW

File: skills/pi-rpc/scripts/cmd/pi-cli/session_test.go [hunk 1]
```diff
@@ -84,6 +84,37 @@ func TestRunSessionCreateUnreachableServer(t *testing.T) {
...
+
+	if err := runSessionCreate(context.Background(), srv.URL, "", "", "/tmp", "", 0); err != nil {
+		t.Fatalf("runSessionCreate with empty provider/model failed: %v", err)
+	}
+	if receivedProvider != "" {
+		t.Errorf("provider forwarded to server = %q, want empty string", receivedProvider)
+	}
+	if receivedModel != "" {
+		t.Errorf("model forwarded to server = %q, want empty string", receivedModel)
+	}
```
[the remark this answers is not in the document]
Comment (rudolfjs) [reply skills/pi-rpc/scripts/cmd/pi-cli/session_test.go:119]: All three points addressed in 83cc963:

- **Data race**: added `done := make(chan struct{})` with `defer close(done)` at the top of the handler; test blocks on `<-done` after `runSessionCreate` returns before reading `receivedProvider`/`receivedModel`.
- **Panic risk**: added `return` after `t.Errorf("decode request body: %v", err)` so a nil `req` is never dereferenced.
- **Portability**: replaced hardcoded `/tmp` with `t.TempDir()`.

Verified with `go test -race -count=1 ./skills/pi-rpc/scripts/cmd/pi-cli/...` — clean pass, no race warnings.

Kept two separate named vars (`receivedProvider`, `receivedModel`) rather than `received[2]string` to match the assertion style used elsewhere in this file.
